packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -432,6 +443,8 @@ export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget
                 connection.listen();
                 if (waitForConnection) {
                     waitForConnection.resolve(connection);
+                    this.isInteractiveTerminal = this.restored;
```

Comment:
I don't understand it. `restored` means that a widget was restored on reload, why `interactive` should be assigned to it?